### ISSUE TITLE
invariant_booleans: false positive on comparisons

### ISSUE THREAD

**Issue by pschiffmann** [dart-lang/linter#1059]
invariant_booleans reports the `else if` condition of the following code:
```dart
void f(int a, int b) {
  if (a < b) {
  } else if (b < a) {}
}
```

whereas this is correctly not reported:
```dart
void f(int a, int b) {
  if (a < b) {
  } else if (a > b) {}
}
```

I'm using the linter that is bundled with Dart 2.0.0-dev.66.0.

**Comment by srawlins** [comment]
Duplicate of https://github.com/dart-lang/linter/issues/914?

**Comment by pschiffmann** [comment]
> Duplicate of #914?

Yes.
